### DISCUSSION TITLE
Using GitHub Releases (possibly auto-generated)?

### DISCUSSION THREAD

**Discussion by LeoniePhiline** [BurntSushi/jiff#152] [Q&A] [answered]
GitHub implements a custom _"Watch repository"_ mode _"Releases"_.
I personally like to use this feature to get notified when features or fixes I am interested in are released and ready to use.

Releases are also a nice way to quickly find out about versions and changes in a central place.

Might it be worth the effort to publish GitHub Releases - ideally automated (for a one-time effort, rather than a recurring one)?

[`release-plz`](https://release-plz.ieni.dev/docs) (which I personally have not yet worked with) promises

- workspace compatibility, 
- integration of `cargo-semver-checks`, 
- changelog generation with `git-cliff`
- and the automated generation of GitHub Releases.

However, of course this kind of automation also comes with a cost:

- setup efforts
- requirement of conventional commits for changelog generation and determining the next semver version
  - needs commit message lints in CI, might increase churn, but also increases consistency, which reduces mental overhead

If releases are rare, then manually creating Releases might be an option for some.
However, it appears to me that you are doing more work than humanly possible already now.
Therefore, I would like to express that this is a very humble request, and I feel a little wary about even bringing this up.
After some consideration, I decided to bring it up, just so you can see there is some very humble interest in GitHub Releases for the project.

**Comment by BurntSushi** [comment, marked as answer]
I use the same release process for Jiff as I do with my dozens of other libraries. I am very aware of tooling like `release-plz` and `cargo-semver-checks` and automatic changelog generation. My process is very lightweight: I manually update the CHANGELOG and do the release locally. I specifically do things this way because I don't have the time to invest in CI to do it for me. I basically just don't trust other tools to get this completely right for me, and when they inevitably break, debugging them in CI is _extremely_ costly for me.

[This is my release script.](https://github.com/BurntSushi/dotfiles/blob/d7b07ed368951b8ae5b171cc450e27f388cf8dc1/bin/cargo-up) In theory, it seems like I could modify this without much fuss to add a `gh` command or two that publishes a new release with the corresponding tag that it creates. I think that would satisfy your particular desire for release notifications, right?

**Reply by LeoniePhiline** [reply]
Thank you for sharing your release script!

> In theory, it seems like I could modify this without much fuss to add a gh command or two that publishes a new release with the corresponding tag that it creates. I think that would satisfy your particular desire for release notifications, right?

It would! 

My primary use case is to get notified at all.
For the release to contain the version's change log excerpt is secondary.

I imagine that `gh release create <tag> --generate-notes` should do it?

**Reply by BurntSushi** [reply]
I'll have to look at what `--generate-notes` does, but yeah.

**Reply by LeoniePhiline** [reply]
As far as I am aware, it generates a link to the tag's commits list (since the previous tag).
